src/Database/DatabaseInitializer.php

```diff
@@ -8,12 +8,16 @@
 use Tempest\Container\Container;
 use Tempest\Container\Initializer;
 use Tempest\Container\Singleton;
+use Tempest\Database\Transactions\TransactionManager;
 
 #[Singleton]
 final readonly class DatabaseInitializer implements Initializer
 {
     public function initialize(Container $container): Database
     {
-        return new GenericDatabase($container->get(PDO::class));
+        return new GenericDatabase(
+            $container->get(PDO::class),
+            $container->get(TransactionManager::class),
```

Comment:
Is this something we can really resolve from the container? What happens if we are interacting with multiple databases and one of them is not supported by PDO?

Reply:
Well right now, we only support PDO for databases, and I don't plan on adding anything else soon. So I'd say it's fine for now

Reply:
@brendt - that's not a very useful mentality for the users of the framework. 😛 

You might not support it out of the box, but interfaces exist so I could add the implementation for an API based database, for example. I have an application in the wild that has two different connections and one is not supported by PDO.

Reply:
In that case I'd say the framework doesn't support it for now and you'll have to manage that other connection manually. 

In an ideal world we support everything from the get-go, but trying to do so would prevent us from ever reaching 1.0. While I agree with you in theory, I think extending the design of the database class outside of PDO is something we'll have to consider after 1.0.